library/common/extensions/key_value/platform/config.cc

```diff
@@ -0,0 +1,52 @@
+#include "library/common/extensions/key_value/platform/config.h"
+
+#include "envoy/config/common/key_value/v3/config.pb.h"
+#include "envoy/config/common/key_value/v3/config.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace KeyValue {
+
+PlatformKeyValueStore::PlatformKeyValueStore(Event::Dispatcher& dispatcher,
+                                             std::chrono::milliseconds flush_interval,
+                                             PlatformInterface& platform_interface,
+                                             const std::string& key)
+    : KeyValueStoreBase(dispatcher, flush_interval), platform_interface_(platform_interface),
+      key_(key) {
+  const std::string contents = platform_interface.read(key);
+  if (!parseContents(contents, store_)) {
+    ENVOY_LOG(warn, "Failed to parse key value store contents {}", key);
+  }
+}
+
+void PlatformKeyValueStore::flush() {
+  std::string output;
+  for (const auto& it : store_) {
+    absl::StrAppend(&output, it.first.length(), "\n", it.first, it.second.length(), "\n",
+                    it.second);
+  }
+  platform_interface_.flush(key_, output);
+}
+
+KeyValueStorePtr
+PlatformKeyValueStoreFactory::createStore(const Protobuf::Message& config,
+                                          ProtobufMessage::ValidationVisitor& validation_visitor,
+                                          Event::Dispatcher& dispatcher, Filesystem::Instance&) {
+  const auto& typed_config = MessageUtil::downcastAndValidate<
+      const ::envoy::config::common::key_value::v3::KeyValueStoreConfig&>(config,
+                                                                          validation_visitor);
+  const auto file_config = MessageUtil::anyConvertAndValidate<
+      envoymobile::extensions::key_value::platform::PlatformKeyValueStoreConfig>(
+      typed_config.config().typed_config(), validation_visitor);
+  auto milliseconds =
+      std::chrono::milliseconds(DurationUtil::durationToMilliseconds(file_config.flush_interval()));
```

Comment:
While I understand the value in supporting periodic flushes, would it make sense for the base case to be synchronous/immediate?

Reply:
I don't think we want to for any of the current use cases - flushing on every dns resolution or every response headers seems expensive.  @RyanTheOptimist for thoughts

Reply:
Yeah for a per-request or per-DNS resolution store, immediate sync would be too expensive. Chrome's analogous system uses periodic writes, fwiw. I could imagine that we might have other use cases for the key-value store for which immediate write might be desirable so maybe there is value in having that as an option. But for the uses we currently have, periodic flushes make sense to me.

Reply:
Okay that's fine - you all are the primary users of this right now anyways. I just thought it might make for easier testing and/or open up other simple use cases that aren't as perf sensitive.